var/spack/repos/builtin/packages/py-torch/package.py

```diff
@@ -235,15 +243,19 @@ def enable_or_disable(variant, keyword='USE', var=None, newer=False):
         env.set('PYTORCH_BUILD_VERSION', self.version)
         env.set('PYTORCH_BUILD_NUMBER', 0)
 
-        # BLAS to be used by Caffe2. Can be MKL, Eigen, ATLAS, or OpenBLAS.
+        # BLAS to be used by Caffe2
         if '^mkl' in self.spec:
             env.set('BLAS', 'MKL')
-        elif '^eigen' in self.spec:
-            env.set('BLAS', 'Eigen')
```

Comment:
Now that `eigen` is a required dependency, it will always be in the spec. Moving it to the bottom to fix ATLAS and OpenBLAS builds, and adding support for vecLib and FLAME builds.